core/src/term/pattern/compile.rs

```diff
@@ -185,26 +185,25 @@ impl CompilePart for RecordPattern {
     //
     //   in
     //
-    //   <if self.tail is empty>
-    //     # if tail is empty, check that the value doesn't contain extra fields
-    //     if final_bindings_id == null ||
-    //        (%static_access% <REST_FIELD> final_bindings_id) != {} then
-    //       null
-    //     else
-    //       %record_remove% "<REST>" final_bindings_id
-    //   <else if self.tail is capture(rest)>
-    //   # move the rest from REST_FIELD to rest, and remove REST_FIELD
-    //     if final_bindings_id == null then
-    //       null
-    //     else
-    //       %record_remove% "<REST>"
-    //         (%record_insert% <rest>
-    //           final_bindings_id
-    //           (%static_access% <REST_FIELD> final_bindings_id)
-    //         )
-    //   <else if self.tail is open>
-    //     %record_remove% "<REST>" final_bindings_id
-    //   <end if>
+    //    if final_bindings_id == null then
+    //      null
+    //    else
```

Comment:
The interesting part of this change is here: it was missing a null check which could lead at runtime to `%record_remove%` complaining that its argument is null on valid (albeit unmatching) patterns. The rest of the diff in this file is just the implementation of this change.